hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -951,24 +899,36 @@ private static Object rewriteRecordWithNewSchemaInternal(Object oldRecord, Schem
         for (int i = 0; i < fields.size(); i++) {
           Schema.Field field = fields.get(i);
           String fieldName = field.name();
+          if (skipMetadataFields && isMetadataField(fieldName)) {
+            if (field.defaultVal() instanceof JsonProperties.Null) {
+              newRecord.put(i, null);
+            } else {
+              newRecord.put(i, field.defaultVal());
+            }
+            continue;
```

Comment:
does the code ever `continue` if `skipMetadataFields && isMetadataField(fieldName)` is true? Looks like it wil execute one of the if-else blocks and never skip.

Reply:
Not sure what you mean. Just put the rest of the stuff into an else block to make it less confusing